File: app/src/main/java/com/example/annada/eventfinder/model/VenueModel.java
package com.example.annada.eventfinder.model;

import android.support.annotation.Nullable;

import org.json.JSONArray;
import org.json.JSONException;

import java.util.List;

/**
 * Created by : annada
 * Date : 21/10/2017.
 */

public class VenueModel {
    private String venueName;
    private String venueRating;
    private String venueIcon;
    private final static String venueIconSize = "64";

    public VenueModel(String venueName, String venueRating, List<Category> venueCategories) {
        this.venueName = venueName;
        this.venueRating = venueRating;
        this.venueIcon = buildIconUri(venueCategories);
    }

    private String buildIconUri(List<Category> venueCategories) {
        return venueCategories.get(0).getIcon().getPrefix() + venueIconSize + venueCategories.get(0).getIcon().getSuffix();
    }

    public String getVenueName() {
        return venueName;
    }

    public String getVenueRating() {
        return venueRating;
    }


    public String getVenueIcon() {
        return venueIcon;
    }

    @Nullable
    public int color;

}

File: app/src/main/java/com/ags/annada/eventfinder/model/VenueModel.kt
package com.ags.annada.eventfinder.model

import org.json.JSONArray
import org.json.JSONException

/**
 * Created by : annada
 * Date : 21/10/2017.
 */

class VenueModel(val venueName: String, val venueRating: String, venueCategories: List<Category>) {
    val venueIcon: String

    var color: Int = 0

    init {
        this.venueIcon = buildIconUri(venueCategories)
    }

    private fun buildIconUri(venueCategories: List<Category>): String {
        return venueCategories[0].icon!!.prefix + venueIconSize + venueCategories[0].icon!!.suffix
    }

    companion object {
        private val venueIconSize = "64"
    }

}
